### PR TITLE
Fixed bug in checksum when entropy is more than 256 bits

### PR DESCRIPTION
**Pull request by rubensayshi**
the hash is sliced and zero padded in such a way that it will only work for entropy up to 256 bits (a checksum of max 8 bits).

PR in 2 parts; first a failing test and then a fix

### CODE REVIEW

File: lib/mnemonic.js [hunk 1]
```diff
@@ -251,4 +246,23 @@ Mnemonic._entropy2mnemonic = function(entropy, wordlist) {
...
+ * @private
+ */
+Mnemonic._entropyChecksum = function(entropy) {
+  var hash = Hash.sha256(entropy);
+  var bits = entropy.length * 8;
+  var cs = bits / 32;
+
+  var hashbits = parseInt(hash.toString('hex'), 16).toString(2);
+  // zero pad the hash bits
+  hashbits = (new Array(256).join("0") + hashbits).slice(-256).slice(0, cs);
```
Comment (rubensayshi) [inline lib/mnemonic.js]: there should be a better way of doing this... but idk how :(

**Comment by maraoz** [comment]
Thanks for your contribution!! :)
can you rebase?

**Comment by rubensayshi** [comment]
sure, is there any way of requiring this module this module in my package.json like `"bitcore-mnemonic": "git://github.com/blocktrail/bitcore-mnemonic.git#8c10f132d1899e2673c46aea898f2a59e9476306"` ?

seems you do some building of some stuff before you push it to npm ?

**Comment by yemel** [comment]
Awesome, thanks for the contribution!
Tests have been fixed on master, could you please rebase one more time :)

**Comment by yemel** [comment]
Also, the constructor has a validation for entropy not being more than 256:

```
if (ent % 32 !== 0 || ent < 128 || ent > 256) {
    throw new errors.InvalidArgument('ENT', 'Values must be 128 < ENT < 256 and ENT % 32 == 0');
  }
```

I think removing the upper bound should be fine. Could you please update that too?

**Comment by rubensayshi** [comment]
yea, I ran into the bug trying to use an mnemonic generated by another library

**Comment by rubensayshi** [comment]
still fails or something that seems unrelated

**Comment by yemel** [comment]
The test was failing because a test exceeded mocha default timeout. I extended it and merge it on this other pull request: https://github.com/bitpay/bitcore-mnemonic/pull/4

Thanks again for the contribution. If you are using bitcore for a project let us know! We may include it on http://bitcore.io

**Comment by rubensayshi** [comment]
great! 

We've been using bitcore for a while now for www.blocktrail.com for the part of our system that is connected to the P2P network and stores block and TX data in our database (inspired by insight).

Unfortunately it is (and will remain) closed source, but if you're listing closed source projects too then it would be cool to be listed :)

**Comment by maraoz** [comment]
That's great to know! I think we should definitely list the project on bitcore.io.

**Pull request merged**